version: "3.1"

networks:
  traefik:
    external: true

services:
  api:
    image: "wexel/auth:api${IMAGE_TAG:-}"
    environment:
      PORT: '80'
      DATABASE_URL: "${DATABASE_URL}"
      PRIVATE_KEY_PATH: '/keys/private.pem'
      PUBLIC_KEY_PATH: '/keys/public.pem'
    volumes:
      - /opt/docker/volumes/keys:/keys
    networks:
      - traefik
    labels:
      - "traefik.http.middlewares.strip.stripprefix.prefixes=/auth"
      - "traefik.http.routers.auth.middlewares=strip@docker"
      - "traefik.http.routers.auth.rule=PathPrefix(`/auth`)"
      - "traefik.http.routers.auth.entrypoints=https"
      - "traefik.http.routers.auth.tls.certresolver=myresolver"
    deploy:
      restart_policy:
        condition: on-failure

  db-init:
    image: "wexel/auth:db-init${IMAGE_TAG:-}"
    environment:
      DB_HOST: "${DB_HOST}"
      DB_NAME: 'auth'
      DB_PASS: "${DB_PASS}"
      DB_PORT: '5432'
      DB_USER: "${DB_USER}"
    deploy:
      restart_policy:
        condition: none

  keygen:
    image: "wexel/auth:keygen${IMAGE_TAG:-}"
    environment:
      PASSPHRASE: ''
      PRIVATE_KEY_PATH: '/keys/private.pem'
      PUBLIC_KEY_PATH: '/keys/public.pem'
    volumes:
      - /opt/docker/volumes/keys:/keys
    deploy:
      restart_policy:
        condition: none
